gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -272,10 +281,17 @@ def setup_ssh_cluster_with_retries(self) -> SSHCluster:
             )
         return cluster
 
-    def run(self) -> GtsfmData:
+    def run(self, graph_partitioner: GraphPartitionerBase = None) -> GtsfmData:
         """Run the SceneOptimizer."""
         start_time = time.time()
 
+        # Create graph partitioner if not provided
+        if graph_partitioner is None:
+            if self.parsed_args.graph_partitioner == "single":
+                from gtsfm.graph_partitioner.single_partition import SinglePartition
+                graph_partitioner = SinglePartition()
```

Comment:
we should make this configurable from the config, but can be done in a separate PR.